cuebot/src/main/java/com/imageworks/spcue/dispatcher/FrameCompleteHandler.java

```diff
@@ -395,13 +396,25 @@ else if (report.getHost().getNimbyLocked()) {
                     && randomNumber.nextInt(100) <= Dispatcher.UNBOOK_FREQUENCY
                     && System.currentTimeMillis() > lastUnbook.get()) {
 
+                // First make sure all jobs have their min cores
+                // Then check for higher priority jobs
+                // If not, rebook this job
                 if (job.autoUnbook && proc.coresReserved >= 100) {
                     if (jobManager.isOverMinCores(job)) {
                         try {
 
                             boolean unbook =
                                     dispatchSupport.findUnderProcedJob(job, proc);
 
+                            if (!unbook) {
+                                JobDetail jobDetail = jobManager.getJobDetail(job.id);
+                                logger.warn("CHECKING Priority...");
```

Comment:
`warn` is probably too high here, right?

Reply:
Ah yes! done.